lte/gateway/c/core/oai/test/spgw_task/test_spgw_procedures.cpp

```diff
@@ -1455,5 +1456,494 @@ TEST_F(SPGWAppProcedureTest, TestDeleteBearerCommand) {
+  sgw_eps_bearer_entry_wrapper_t* spgw_eps_bearer_entry_p =
+      LIST_FIRST(pgw_ni_cbr_proc->pending_eps_bearers);
+  teid_t ue_ded_bearer_sgw_teid =
+      spgw_eps_bearer_entry_p->sgw_eps_bearer_entry->s_gw_teid_S1u_S12_S4_up;
+
+  // send bearer activation response from MME
+  ebi_t ded_eps_bearer_id = DEFAULT_EPS_BEARER_ID + 1;
+  itti_s11_nw_init_actv_bearer_rsp_t sample_nw_init_ded_bearer_actv_resp = {};
+  fill_nw_initiated_activate_bearer_response(
+      &sample_nw_init_ded_bearer_actv_resp, DEFAULT_MME_S11_TEID, ue_sgw_teid,
+      ue_ded_bearer_sgw_teid, DEFAULT_ENB_GTP_TEID + 1, ded_eps_bearer_id,
+      REQUEST_ACCEPTED, test_plmn);
+  return_code = sgw_handle_nw_initiated_actv_bearer_rsp(
+      &sample_nw_init_ded_bearer_actv_resp, test_imsi64);
+
+  EXPECT_EQ(return_code, RETURNok);
+
+  // check that bearer is created
+  EXPECT_TRUE(is_num_s1_bearers_valid(ue_sgw_teid, 2));
+
+  // check that no pending procedure is left
+  EXPECT_EQ(
+      get_num_pending_create_bearer_procedures(
+          &spgw_eps_bearer_ctxt_info_p->sgw_eps_bearer_context_information),
+      0);
+
+  // send deactivate request for dedicated bearer from Session Manager
+  // with invalid imsi
+  itti_gx_nw_init_deactv_bearer_request_t
+      sample_gx_nw_init_ded_bearer_deactv_req = {};
+  fill_nw_initiated_deactivate_bearer_request(
+      &sample_gx_nw_init_ded_bearer_deactv_req, invalid_imsi_str,
+      DEFAULT_EPS_BEARER_ID, ded_eps_bearer_id);
+
+  // check that MME does not get bearer deactivation request
+  EXPECT_CALL(
+      *mme_app_handler,
+      mme_app_handle_nw_init_bearer_deactv_req(
+          check_params_in_deactv_bearer_req(
+              sample_gx_nw_init_ded_bearer_deactv_req.no_of_bearers,
+              sample_gx_nw_init_ded_bearer_deactv_req.ebi)))
+      .Times(0);
+
+  return_code = spgw_handle_nw_initiated_bearer_deactv_req(
+      &sample_gx_nw_init_ded_bearer_deactv_req, test_imsi64);
+
+  EXPECT_EQ(return_code, RETURNok);
+
+  // check that there 2 bearers
+  EXPECT_TRUE(is_num_s1_bearers_valid(ue_sgw_teid, 2));
+
+  // send deactivate request for dedicated bearer from Session Manager
+  // with invalid imsi
+  fill_nw_initiated_deactivate_bearer_request(
+      &sample_gx_nw_init_ded_bearer_deactv_req, invalid_imsi_str,
+      DEFAULT_EPS_BEARER_ID, ded_eps_bearer_id);
+
+  // check that MME does not get bearer deactivation request
+  EXPECT_CALL(
+      *mme_app_handler,
+      mme_app_handle_nw_init_bearer_deactv_req(
+          check_params_in_deactv_bearer_req(
+              sample_gx_nw_init_ded_bearer_deactv_req.no_of_bearers,
+              sample_gx_nw_init_ded_bearer_deactv_req.ebi)))
+      .Times(0);
+
+  return_code = spgw_handle_nw_initiated_bearer_deactv_req(
+      &sample_gx_nw_init_ded_bearer_deactv_req, test_imsi64);
+
+  EXPECT_EQ(return_code, RETURNok);
+
+  // verify that the dedicated bearer is not deleted
+  EXPECT_TRUE(is_num_s1_bearers_valid(ue_sgw_teid, 2));
```

Comment:
This is repeat of lines 1735-1755

Reply:
The first call is with invalid IMSI and the second one is with invalid LBI. In both cases call flow will be same

Reply:
Lines 1783 - 1804 are for invalid LBI.
Lines 1735 - 1755 are for invalid IMSI
Lines 1760 - 1781 are also for invalid IMSI , which is a duplicate